core/src/test/java/org/apache/iceberg/data/DataTest.java

```diff
@@ -148,7 +148,11 @@ public void testTypeSchema(Type type) throws IOException {
         .as("variant is not yet implemented")
         .isTrue();
 
-    writeAndValidate(new Schema(required(1, "id", LongType.get()), optional(2, "test_type", type)));
+    writeAndValidate(
+        new Schema(
+            required(1, "id", LongType.get()),
+            optional(2, "test_type", type),
+            required(3, "trailing_data", Types.StringType.get())));
```

Comment:
Without correct alignment, this test will fail for unknown when the `null` value for field 2 is passed to the writer for field 3.